src/ext.rs

```diff
@@ -832,6 +858,31 @@ fn timeout2ms(dur: DWORD) -> Option<u32> {
 }
 
 #[cfg(any(feature = "nightly", feature = "duration"))]
+fn linger2dur(linger_opt: linger) -> Option<Duration> {
+    let dur = Duration::from_secs(linger_opt.l_linger as u64);
+    if dur.as_secs() == 0 {
+        None
+    }
+    else {
+        Some(dur)
+    }
+}
+
+#[cfg(any(feature = "nightly", feature = "duration"))]
+fn dur2linger(dur: Option<Duration>) -> linger {
+    match dur {
+        Some(d) => {
+            let onoff = if d.as_secs() == 0 { 0 } else { 1 };
+            linger {
+                l_onoff: onoff as u16,
```

Comment:
I changed this, and added a second implementation of dur2linger, which just changes the type conversation from u16 to c_int. On my cloud9 machine I've got this working with an extended libc.

Is there anything I can do to make this look better? Something like `... as if cfg!(windows) {u16} else {c_int}` doesn't work.